view/frontend/requirejs-config.js

```diff
@@ -49,13 +50,16 @@ const config = {
         'algoliaMustacheLib': 'Algolia_AlgoliaSearch/js/lib/mustache.min',
         'algoliaHoganLib'   : 'Algolia_AlgoliaSearch/js/lib/hogan.min',
 
+        // Other plugins
+        'algoliaLocalStorageRecentSearchesLib'  : 'Algolia_AlgoliaSearch/js/lib/local-storage-recent-searches-plugin.min',
```

Comment:
This should be an Autocomplete plugin - like what we've [included](https://github.com/algolia/algoliasearch-custom-algolia-magento-2/blob/master/view/frontend/web/js/internals/autocomplete-plugin-recent-searches.js) with `CustomAlgolia`.